integrations/access/email/mailers.go

```diff
@@ -147,3 +231,29 @@ func (m *MailgunMailer) Send(ctx context.Context, id, recipient, body, reference
 
 	return id, nil
 }
+
+// statusSinkTransport wraps the Mailgun client transport and
+// emits plugin status.
+type statusSinkTransport struct {
+	http.RoundTripper
+	sink common.StatusSink
+}
+
+// RoundTrip implements the http.RoundTripper interface.
+func (t *statusSinkTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	log := logger.Get(req.Context())
+	resp, err := t.RoundTripper.RoundTrip(req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if t.sink != nil {
+		ctx, cancel := context.WithTimeout(req.Context(), statusEmitTimeout)
+		defer cancel()
+
+		status := common.StatusFromStatusCode(resp.StatusCode)
```

Comment:
@bernardjkim 

sorry for the delay. A few months ago, we began adding dedicated status entries for plugins, where we include detailed error information like the error code, a human-readable message, and the original error itself. We’ve implemented this approach for Entra, GitLab, and others.

I suggest doing the same here. A simple "running/not running" status doesn’t provide meaningful insight, especially when running plugins in teleport cloud tenants where users don't have direct access to logs.

Examples of good information:
- can't dial to mailgun
- credentials invalid (expired)
- message rejected...

check 

https://github.com/gravitational/teleport/blob/64922dffc15846a395f765fc524988227e6a3bc2/api/types/types.pb.go#L17107-L17130

Reply:
Hey, that is a good point. The generic "running / not running" status isn't all that helpful. I'll work on a follow up PR soon to improve the status entries.

Reply:
Tracking in https://github.com/gravitational/teleport/issues/48604 so I don't forget